pkg/shifter/shifter.go

```diff
@@ -0,0 +1,255 @@
+package shifter
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"sync"
+	"time"
+
+	"github.com/roppenlabs/silent-assassin/pkg/config"
+	"github.com/roppenlabs/silent-assassin/pkg/gcloud"
+	"github.com/roppenlabs/silent-assassin/pkg/k8s"
+	"github.com/roppenlabs/silent-assassin/pkg/killer"
+	"github.com/roppenlabs/silent-assassin/pkg/logger"
+	"github.com/roppenlabs/silent-assassin/pkg/notifier"
+	container "google.golang.org/api/container/v1"
+	v1 "k8s.io/api/core/v1"
+)
+
+const timeLayout string = "15:04"
+
+//Suggest a better name
+type npShiftConf struct {
+	destNodePool       string
```

Comment:
Can we call them onDemandNodePool and preemptibleNodePool ? I can see source and destination also being used. Let's stick to one convention. A little hard to understand when there are multiple ways of addressing the same.